### ISSUE TITLE
Nahrávání souborů SHP

### ISSUE THREAD

**Issue by tomvojtechovsky** [ARUP-CAS/aiscr-gis-convert#1]
Pokusit se implementovat nahrávání souborů pouze zadáním souboru `.shp`.
Všechny ostatní přípony skrýt.

**Comment by tomvojtechovsky** [comment]
První ohledání řešení neúspěšné.

**Comment by motyc** [comment]
@tomvojtechovsky Nevadí. Pokud to nepůjde, přežijeme to. Tady ale třeba víme, že to tak funguje: https://www.km4city.org/wkt/ nebo https://mygeodata.cloud/converter/

**Comment by tomvojtechovsky** [comment]
Děkuji za zdroje.
Nicméně mi nejde ani na jednom nahrát pouze soubor.shp a načíst ho - co dělám špatně?

 https://www.km4city.org/wkt/

![image](https://github.com/user-attachments/assets/b2116159-8ec3-4a7d-a5ba-84e6a0be71d4)

===>>

![image](https://github.com/user-attachments/assets/4ef32f14-eba2-41f6-a201-869ac4d8cf22)

**Comment by motyc** [comment]
@tomvojtechovsky Problém je v labelu - když zadám "test", hlásí to totéž. Když zadám "aaa", funguje to. Zjevně nějaký bug. S tím výběrem souboru to ale nesouvisí.

**Comment by tomvojtechovsky** [comment]
Ano, už mi to také funguje. Děkuji. Podívám se na to :)

**Comment by tomvojtechovsky** [comment]
Prohlédl jsem zdrojový kód https://www.km4city.org/wkt/ a aplikovat to. Nicméně ani po hledání nějaké funkční cesty k řešení se to nepodařilo. Ještě to zkusím později znovu, ale nyní bych to nechal tak, že uživatel musí vybrat všechny čtyři soubory.

Případné dovylepšení může být, že pokud uživatel nezadá všechny potřebné soubory, bude vyzván k dooznačení zbylých souborů, dokud nebudou načtené všechny čtyři potřebné. Přijde mi to trochu jako overkill řešení a možná také trochu zmatečné. Každopádně by to vedlo k tomu, aby i "lamy" byly nuceni vybrat (ačkoli postupně) všechny soubory.

**Comment by motyc** [comment]
@tomvojtechovsky Nebo by možná stačila prozatím kontrola, že označené soubory si odpovídají názvy... Jde o to ošetřit tu výjimku, která může nastat, když to někdo udělá chybně.

**Comment by tomvojtechovsky** [comment]
Ano, kontrolu přidám. Zamyslím se, jak donutit uživatele označit všechny potřebné soubory a pokud tak neudělá, aby to věděl.

Mám osobně testovací adresář s shp - 4 × 6 souborů. A riziko chybného označení je relativně veliké :/
I proto mě stále láká myšlenka, že by uživatel mohl teoreticky označit v podstatě jakékoli soubory - třeba i všechno, co má v adresáři. Pokud by výběr obsahoval nějakou kompletní sadu souborů, bylo by možné data nahrát.
 - pokud by výběr obsahoval jednu sadu shapefile - automaticky by mohla být nahrána
 - pokud by výběr obsahoval více shapefilů - dostal by na výběr
 - teoreticky může být nahráno více shapefilů, které budou označeny - ale tím si značně zkomplikujeme UI při výběru epsg a atributu pro label. - Tomuto řešení bych se zatím vyhnul.

Popravdě, toto je docela k zamyšlení. Realizace by nemusela být složitá a zdlouhavá. Asi nechceme vytvářet kanón na vrabce, ale pokud se nepovede načítání pouze pomocí souboru.shp (a zbytek automaticky), pak bych se k nějakému robustnějšímu řešení přikláněl.

**Comment by motyc** [comment]
Navrhuji zatím dotáhnout ostatní věci a toto pak dořešíme. Zejména to musí být pro uživatele co nejvíce srozumitelné a předvídatelné. Určitě to nějak vhodně vymyslíme. Dobré by pro začátek bylo, aby to vždy ověřilo, že to dostalo vše potřebné a případně to zahlásilo chybu.

**Comment by tomvojtechovsky** [comment]
Ok, prozatím odloženo. 
Pro kontrolu souborů je samostatné issue #11
